src/general.tsx

```diff
@@ -60,30 +50,16 @@ function notifyOtherLanguages(preferred_language: string, website_language: stri
     );
 }
 
-// This uses the `navigator.language` property (similar-ish to the `Accept-Language` header which we cannot access from
-// JS) which may not necessarily represent the user's country (or even include region information at all).
-// The more reliable way would be to feed the user's IP into a geolocation service but that is not an option, so we have
-// to stick with this.
-function guessUserCountry(): typeof window.globals.country {
-    const navigator_lang = navigator.language;
-
-    // Taken from: https://github.com/gagle/node-bcp47/blob/a74d98d43d16b0094b2f4ea8e7a58f4b5830c15b/lib/index.js#L4
-    const bcp47_regex =
-        /^(?:(en-gb-oed|i-ami|i-bnn|i-default|i-enochian|i-hak|i-klingon|i-lux|i-mingo|i-navajo|i-pwn|i-tao|i-tay|i-tsu|sgn-be-fr|sgn-be-nl|sgn-ch-de)|(art-lojban|cel-gaulish|no-bok|no-nyn|zh-guoyu|zh-hakka|zh-min|zh-min-nan|zh-xiang))$|^((?:[a-z]{2,3}(?:(?:-[a-z]{3}){1,3})?)|[a-z]{4}|[a-z]{5,8})(?:-([a-z]{4}))?(?:-([a-z]{2}|\d{3}))?((?:-(?:[\da-z]{5,8}|\d[\da-z]{3}))*)?((?:-[\da-wyz](?:-[\da-z]{2,8})+)*)?(-x(?:-[\da-z]{1,8})+)?$|^(x(?:-[\da-z]{1,8})+)$/i;
-    const bcp47_country = (bcp47_regex.exec(navigator_lang)?.[5] || '').toLowerCase() as typeof window.globals.country;
-
-    const bcp47_preferred_language = (bcp47_regex.exec(navigator_lang)?.[3] || '').toLowerCase();
-    const bcp47_website_language = (bcp47_regex.exec(document.documentElement.lang)?.[3] || '').toLowerCase();
-
-    if (bcp47_preferred_language !== bcp47_website_language && bcp47_preferred_language in fallback_countries) {
-        notifyOtherLanguages(bcp47_preferred_language, bcp47_website_language);
-    }
+if (!useAppStore.getState().country) {
+    // TODO: Remove the cookie migration code in a year or so.
+    useAppStore.getState().changeCountry((Cookie.get('country') as Country) || guessUserCountry());
+    Cookie.remove('country');
 
-    // If we cannot guess the user's country, it makes sense to fallback to the language.
-    if (!navigator_lang || !bcp47_country) return fallback_countries[window.LOCALE];
+    const { language: preferred_language } = parseBcp47Tag(navigator.language);
+    const { language: website_language } = parseBcp47Tag(document.documentElement.lang);
 
-    // If however we *can* guess the country but just don't support it, we show all companies.
-    return window.SUPPORTED_COUNTRIES.includes(bcp47_country) ? bcp47_country : 'all';
+    if (preferred_language !== website_language && preferred_language! in fallback_countries)
```

Comment:
This is probably just a staging bug, but If I change languages in the netlify build preview, I am not shown the choose language modal.

Reply:
Um, I can't reproduce that.

![tFxNwRv8cr](https://user-images.githubusercontent.com/4048809/168801519-e953b6b3-addd-494a-8234-c2a06fd675f4.gif)

Reply:
Ah, no. I am talking about the language suggestion flash.

Reply:
Well yeah. If you change the language on the staging site, you're still on the same origin, so your country setting (which determines whether the suggestion is shown) is preserved.